sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -0,0 +1,867 @@
+                gnrc_netdev_lwmac_set_quit_rx(gnrc_netdev, true);
+            }
+
+            /* Here we check if we are close to the end of the cycle. If yes,
+             * go to sleep. Firstly, get the relative phase. */
+            uint32_t phase = rtt_get_counter();
+            if (phase < gnrc_netdev->lwmac.last_wakeup) {
+                phase = (RTT_US_TO_TICKS(LWMAC_PHASE_MAX) - gnrc_netdev->lwmac.last_wakeup) +
+                         phase;
+            }
+            else {
+                phase = phase - gnrc_netdev->lwmac.last_wakeup;
+            }
+            /* If the relative phase is beyond 4/5 cycle time, go to sleep. */
+            if (phase > (4*RTT_US_TO_TICKS(LWMAC_WAKEUP_INTERVAL_US)/5)) {
+                gnrc_netdev_lwmac_set_quit_rx(gnrc_netdev, true);
+            }
+
+            if (gnrc_netdev_lwmac_get_quit_rx(gnrc_netdev)) {
+                lwmac_set_state(gnrc_netdev, LWMAC_SLEEPING);
+            }
+            else {
+                /* Go back to LISTENING for keep hearing on the channel */
+                lwmac_set_state(gnrc_netdev, LWMAC_LISTENING);
+            }
+            break;
+        }
+        case RX_STATE_SUCCESSFUL: {
+            LOG(LOG_DEBUG, "[lwmac] Reception was successful\n");
+            lwmac_rx_stop(gnrc_netdev);
+            /* Dispatch received packets, timing is not critical anymore */
+            gnrc_mac_dispatch(&gnrc_netdev->rx);
+
+            /* Here we check if we are close to the end of the cycle. If yes,
+             * go to sleep. Firstly, get the relative phase. */
+            uint32_t phase = rtt_get_counter();
+            if (phase < gnrc_netdev->lwmac.last_wakeup) {
+                phase = (RTT_US_TO_TICKS(LWMAC_PHASE_MAX) - gnrc_netdev->lwmac.last_wakeup) +
+                         phase;
+            }
+            else {
+                phase = phase - gnrc_netdev->lwmac.last_wakeup;
+            }
+            /* If the relative phase is beyond 4/5 cycle time, go to sleep. */
+            if (phase > (4*RTT_US_TO_TICKS(LWMAC_WAKEUP_INTERVAL_US)/5)) {
+                gnrc_netdev_lwmac_set_quit_rx(gnrc_netdev, true);
+            }
+
+            if (gnrc_netdev_lwmac_get_quit_rx(gnrc_netdev)) {
+                lwmac_set_state(gnrc_netdev, LWMAC_SLEEPING);
+            }
+            else {
+                /* Go back to LISTENING after successful reception */
+                lwmac_set_state(gnrc_netdev, LWMAC_LISTENING);
+            }
+            break;
+        }
+        default:
+            lwmac_rx_update(gnrc_netdev);
+    }
+
+    /* If state has changed, reschedule main state machine */
+    if (state_rx != gnrc_netdev->rx.state) {
+        lwmac_schedule_update(gnrc_netdev);
+    }
+}
+
+static void _tx_management(gnrc_netdev_t *gnrc_netdev)
+{
+    char *tx_success = "";
+    lwmac_tx_state_t state_tx = gnrc_netdev->tx.state;
+
+    switch (state_tx) {
+        case TX_STATE_STOPPED: {
+            gnrc_pktsnip_t *pkt;
```

Comment:
move to new function `static void _tx_management_stopped(gnrc_netdev_t *gnrc_netdev);`?